Python/bytecodes.c

```diff
@@ -217,61 +217,60 @@ dummy_func(
             LOAD_FAST,
         };
 
-        inst(LOAD_FAST_CHECK, (-- value)) {
+        inst(LOAD_FAST_CHECK, (-- value: _PyStackRef)) {
             value = GETLOCAL(oparg);
-            if (value == NULL) {
+            if (PyStackRef_IsNull(value)) {
                 _PyEval_FormatExcCheckArg(tstate, PyExc_UnboundLocalError,
                     UNBOUNDLOCAL_ERROR_MSG,
                     PyTuple_GetItem(_PyFrame_GetCode(frame)->co_localsplusnames, oparg)
                 );
                 ERROR_IF(1, error);
             }
-            Py_INCREF(value);
+            PyStackRef_INCREF(value);
         }
 
-        replicate(8) pure inst(LOAD_FAST, (-- value)) {
+        replicate(8) pure inst(LOAD_FAST, (-- value: _PyStackRef)) {
             value = GETLOCAL(oparg);
-            assert(value != NULL);
-            Py_INCREF(value);
+            assert(PyStackRef_To_PyObject_Steal(value) != NULL);
```

Comment:
```suggestion
            assert(PyStackRef_To_PyObject_Borrow(value) != NULL);
```

This steals a reference, then drops it.